packages/package_info/lib/package_info.dart

```diff
@@ -24,30 +24,29 @@ class PackageInfo {
   /// See [fromPlatform] for the right API to get a [PackageInfo] that's
   /// actually populated with real data.
   PackageInfo({
-    this.appName,
-    this.packageName,
-    this.version,
-    this.buildNumber,
+    required this.appName,
+    required this.packageName,
+    required this.version,
+    required this.buildNumber,
   });
 
-  static PackageInfo _fromPlatform;
+  static PackageInfo? _fromPlatform;
 
   /// Retrieves package information from the platform.
   /// The result is cached.
   static Future<PackageInfo> fromPlatform() async {
     if (_fromPlatform != null) {
-      return _fromPlatform;
+      return _fromPlatform!;
```

Comment:
shouldn't need `_fromPlatform!` since the analyzer is able to detect the enclosing `_fromPlatform != null` check.

Reply:
So, I was wondering about that too. When I remove it, I get this error 

<img width="795" alt="Screen Shot 2021-01-07 at 4 36 31 PM" src="https://user-images.githubusercontent.com/10687576/103947715-8badaa00-5106-11eb-8cd3-c54d0c2f7671.png">

I also made it synchronous just in case that was the problem. It seems like the static analysis is missing the null check. I'm probably missing something obvious, but have you seen this before?

Reply:
ah yeah. @mehmetf just pointed out that the variable needs to be defined in the local scope